darwin: set the minimum version to macOS 11.3; nixos/doc/rl-2505: announce macOS version support policy

emilazy opened the pull request:
This changes the minimum macOS version to macOS Big Sur 11.3 for 25.05, as announced in https://github.com/NixOS/nixpkgs/pull/338695, and further proposes that future Nixpkgs releases are expected to only guarantee support for versions of macOS that are supported by Apple, starting with macOS Sonoma 14 for Nixpkgs 25.11.

Apple only provides security updates for the last three releases. The statistics I’ve found point towards the conclusion that use of unsupported macOS releases drops off very quickly, especially among technical users:

* [PyPI download stats for Pillow](https://discuss.python.org/t/moving-packaging-and-installers-to-macos-10-13-as-a-minimum/31907/5) show that 98.35% of downloads were for a supported macOS version (macOS ≥ 11) in July–August 2023. 

* The author of [Downie](https://software.charliemonroe.net/downie/), a macOS video downloader application, [shares statistics](https://blog.charliemonroe.net/sunsetting-support-for-older-macos-versions/) showing that 92.35% of users were on a supported macOS version (macOS ≥ 12) in September 2024. In April 2023, the middle of the macOS 13 release cycle, 98.17% of users were on a supported version (macOS ≥ 11); even in December 2022, only two months after the macOS 13 release and around the time a Nixpkgs release would leave support, 97% of users are listed as being on a supported version, although those numbers were from Setapp and so probably cannot be compared one‐to‐one with the later statistics.

* When I was discussing version statistics in the Nix on macOS Matrix room at the start of the month, grahamc posted some charts from Determinate Systems data. I believe that one reflected early user data for their downstream Determinate Nix distribution and one reflected ephemeral CI installations using their installer (though I am not sure whether it was for all installations or only Determinate Nix). These charts are naturally going to be skewed by multiple factors and can’t be used as directly indicative of ongoing Nix on macOS usage, but both indicated that essentially 100% of their users are on a supported OS release, with even macOS 12, the oldest supported release at the time, only showing up in the charts as a small slice of GitHub Actions CI runners. (GitHub themselves are [removing that runner image](https://github.com/actions/runner-images/issues/10721) on about the same schedule we would drop support for it if we were already following this policy.)

Given this data, the fact that users remaining on unsupported OS versions are unlikely to have too much overlap with those who update to the newest Nixpkgs releases in a timely manner, and the unsafety of using macOS releases that are out of security support, I conclude that it’s not a good use of resources for us to support unsupported macOS versions in new Nixpkgs releases, and we should simply bump our requirements on the same schedule as Apple, as Homebrew does. This doesn’t mean that every package will instantly break for users on old versions; just that we’ll build packages with the newer SDK release and deployment target, and accept that some of the resulting binaries may use new APIs that don’t work on unsupported OS releases.

Keeping our supported versions in sync with Apple will result in fewer build failures because of unsupported APIs, reduced need for manual SDK version overrides to address those failures, and more OS features being utilized out of the box.  Some of the FOSS ecosystem is already ahead of us: Qt dropped support for macOS 11 in 6.8, so 25.05 will ship with a minimum supported OS version that hasn’t been able to run the latest Qt release for over half a year. QEMU only officially supports the last two macOS releases, although they mercifully avoided using any features beyond the macOS 12 SDK in the period where we weren’t able to use anything newer. There are always going to be packages that don’t support every macOS version we do, but continuing to lag significantly behind some of the most important software we ship would be counterproductive.

As Apple tends to release new versions (and drop support for the oldest) around September–October, users of the oldest supported macOS release will have two to three months with a supported Nixpkgs version after that macOS release goes out of security support. I think that’s a reasonable buffer given the dynamics of the macOS ecosystem; we’ll still be supporting macOS releases for three years when we only support NixOS releases for seven months. After that, users can install a supported OS (potentially using OpenCore Legacy Patcher for old hardware, or even switching to NixOS), upgrade to the new Nixpkgs release and accept that things may break, continue using the current Nixpkgs release (it won’t get security updates, but neither will their OS), or switch to MacPorts, which keeps support for much older versions.

As far as the 11.3 bump goes, there’s a lot of cruft lying around to deal with the fact that we’ve been stuck on the 10.12 SDK on `x86_64-darwin` for a long time. That can now be cleaned up, reducing the maintenance burden of Darwin support across the project. In this PR, I’ve focused on things required to keep eval working after old SDKs are removed and simple clean‐ups of the SDK and source release packages, the Darwin standard environment, and compiler toolchains. I threw in a Nix change just for kicks, since I introduced that workaround and kind of hate it, and dropped two compatibility hooks that are no longer required. There’s a lot more throughout the tree that can be removed after this – I have almost 100 commits pending in my local branch – but this is a good start.

cc @NixOS/darwin-maintainers

<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
    * `apple-sdk_{11,12,13,14,15}`
    * `darwin.IOKitTools`
    * `darwin.adv_cmds`
    * `darwin.copyfile`
    * `darwin.diskdev_cmds`
    * `darwin.file_cmds`
    * `darwin.libpcap`
    * `darwin.libresolv`
    * `darwin.mail_cmds`
    * `darwin.network_cmds`
    * `darwin.patch_cmds`
    * `darwin.removefile`
    * `darwin.shell_cmds`
    * `darwin.system_cmds`
    * `darwin.text_cmds`
    * `darwin.top`
    * `gcc{10,11,12,13,14}`
    * `gtk4`
    * `haskellPackages.network`
    * `hello`
    * `libsForQt5.{qtbase,qttools,qttranslations,qtimageformats,qtdeclarative,qtmultimedia}`
    * `llvmPackages_{13,14,15,16,17,18,19}.{compiler-rt,libcxx}`
    * `llvmPackages_{15,16,17,18,19}.lldb`
    * `nixVersions.{nix_2_24,git}`
    * `python3`
    * `qt6Packages.qtbase`
    * `tests.cc-wrapper.default`
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [x] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

emilazy committed:
darwin: set the minimum version to macOS 11.3

emilazy committed:
nixos/doc/rl-2505: announce macOS version support policy

emilazy committed:
apple-sdk_{10_12,10_13,10_14,10_15}: drop

These can no longer be used now that the minimum is 11.3.

emilazy committed:
apple-sdk: remove code for old SDKs

emilazy committed:
llvmPackages.compiler-rt: remove code for macOS < 11

emilazy committed:
darwin.top: remove code for macOS < 11

emilazy committed:
darwin.libpcap: remove code for macOS < 11

emilazy committed:
darwin.system_cmds: remove code for macOS < 11

We need to pull in an additional XNU header for a tool we are now
able to build.

emilazy committed:
darwin.text_cmds: remove code for macOS < 11

emilazy committed:
darwin.mail_cmds: remove code for macOS < 11

emilazy committed:
darwin.network_cmds: remove code for macOS < 11

emilazy committed:
darwin.adv_cmds: remove code for macOS < 11

emilazy committed:
darwin.diskdev_cmds: remove code for macOS < 11

emilazy committed:
darwin.removefile: remove code for macOS < 11

emilazy committed:
darwin.file_cmds: remove code for macOS < 11

emilazy committed:
darwin.IOKitTools: remove code for macOS < 11

emilazy committed:
darwin.libresolv: remove code for macOS < 11

emilazy committed:
darwin.shell_cmds: remove code for macOS < 11

emilazy committed:
darwin.copyfile: remove code for macOS < 11

emilazy committed:
darwin.patch_cmds: remove code for macOS < 11

emilazy committed:
python3: remove code for macOS < 11

emilazy committed:
memstream{,Hook}: drop

emilazy committed:
memorymapping{,Hook}: drop

emilazy committed:
stdenv/darwin: remove code for old SDKs

emilazy committed:
llvmPackages.lldb: remove code for macOS < 11

emilazy committed:
llvmPackages.libcxx: remove code for macOS < 11

emilazy committed:
nixVersions.{nix_2_24,git}: remove code for macOS < 11

emilazy committed:
cctools: remove code for macOS < 11

As with ld64, flags added as part of macOS 10.12 compatibility shims
(in this case `-undefined dynamic_lookup`) were inadvertently fixing
the build. The Xcode project links `otool-classic` and `strip` against
`libc++`; why it was previously working on AArch64 without linking
against it or passing `-undefined dynamic_lookup`, I can’t fathom.

emilazy committed:
gcc: remove code for macOS < 11

emilazy committed:
ld64: remove code for macOS < 11

We need to make `-DPRIVATE` unconditional, as it’s required for a
XNU header definition that is only used on `x86_64-darwin`, but was
coincidentally getting set by a now‐obsolete compatibility shim.

emilazy committed:
libsForQt5.qtbase: remove code for macOS < 11

emilazy committed:
haskellPackages.network: remove code for macOS < 11

This reverts commit 2573d499bd7cfb3d1d1f52e20c5b8bf49eb5f7d8.

emilazy committed:
darwin.apple_sdk_10_12: drop

emilazy committed:
gtk4: remove code for macOS < 11

emilazy committed:
ld64: use the SDK source release packages

viraptor commented on doc/stdenv/platform-notes.chapter.md:
```diff
@@ -84,24 +84,19 @@ Some common issues when packaging software for Darwin:
 
   | Xcode version      | SDK version                                       | nixpkgs attribute |
   |--------------------|---------------------------------------------------|-------------------|
-  | Varies by platform | 10.12.2 (x86_64-darwin)<br/>11.3 (aarch64-darwin) | `apple-sdk`       |
```
Should the entry for the bare `apple-sdk` be in this table, just to be clear what does it do after the change?

emilazy replied:
I’ve added the `apple-sdk` reference to the same table entry as `apple-sdk_11` now that it’s not platform-specific; let me know what you think!

viraptor replied:
perfect ❤️

paparodeo commented on pkgs/top-level/darwin-packages.nix:
```diff
@@ -66,12 +66,7 @@ makeScopeWithSplicing' {
-    "11.0" = apple_sdk_11_0;
-  }.${stdenv.hostPlatform.darwinSdkVersion}
-  or (throw "Unsupported sdk: ${stdenv.hostPlatform.darwinSdkVersion}");
+  apple_sdk = apple_sdk_11_0;
```
does this need to be coupled to `darwinSdkVersion` with an assert or something? there wasn't one before I added it and seemed like it just broke things like android and iOS eval.

emilazy replied:
Hmm, since this is now just a compatibility alias that are going away by the next time this changes in 25.11, it seems fine? I struggle to imagine how it could break Android eval if it’s not even accessing the platform.

I thought about adding an assert to `apple-sdk` to make sure that the platform fields are updated to the right minor version when the SDK is bumped, but I couldn’t quite think of a properly elegant way to do it, and since our minimum supported macOS release is going to continue to be years old it seems unlikely they’ll get minor SDK updates very often.

paparodeo replied:
oh yeah, I forget `apple-sdk` is the thing now -- and since that uses `darwinSdkVersion` I guess the assert I was looking for is not needed.

paparodeo commented on pkgs/by-name/cc/cctools/meson.build:
```diff
@@ -553,7 +539,7 @@ install_man('man/strings.1')
     f'@target_prefix@strip',
     c_args : ['-DTRIE_SUPPORT'],
-    dependencies : [libcodedirectory, libprunetrie],
+    dependencies : [libcxx, libcodedirectory, libprunetrie],
```
weird. I only see strip needed to link with lib stuff
https://github.com/apple-oss-distributions/cctools/blob/7a5450708479bbff61527d5e0c32a3f7b7e4c1d0/cctools.xcodeproj/project.pbxproj#L2221-L2228

emilazy replied:
Yeah, so it turns out that the Xcode build system is even more baroque than you can imagine; there’s [another file](https://github.com/apple-oss-distributions/cctools/blob/7a5450708479bbff61527d5e0c32a3f7b7e4c1d0/xcode/strip.xcconfig).

paparodeo commented on pkgs/os-specific/darwin/apple-source-releases/adv_cmds/meson.build.in:
```diff
@@ -4,8 +4,6 @@
 # Project settings
 project('adv_cmds', 'c', 'cpp', version : '@version@')
 
-sdk_version = get_option('sdk_version')
```
can delete the file `meson.options` too

emilazy replied:
Yep, done.

paparodeo reviewed (approved):
LGTM

emilazy commented:
Rebased for conflicts.

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/the-darwin-sdks-have-been-updated/55295/8

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/the-darwin-sdks-have-been-updated/55295/26

siriobalmelli commented:
Can't run nixpkgs-review on x86_64-darwin; if that's expected ignore this dump:

```
$ git -c fetch.prune=false fetch --no-tags --force https://github.com/NixOS/nixpkgs staging:refs/nixpkgs-review/0 pull/352129/head:refs/nixpkgs-review/1
remote: Enumerating objects: 6554, done.
remote: Counting objects: 100% (4434/4434), done.
remote: Compressing objects: 100% (1027/1027), done.
remote: Total 6554 (delta 3801), reused 3822 (delta 3359), pack-reused 2120 (from 1)
Receiving objects: 100% (6554/6554), 5.81 MiB | 8.23 MiB/s, done.
Resolving deltas: 100% (4431/4431), completed with 1205 local objects.
From github.com:NixOS/nixpkgs
   ce92a3fe85c9..2f7285a72969  staging               -> refs/nixpkgs-review/0
 + 35d342567a48...f17e590355d6 refs/pull/352129/head -> refs/nixpkgs-review/1  (forced update)
$ git worktree add /Users/sirio/.cache/nixpkgs-review/pr-352129/nixpkgs 2f7285a7296988838b6e106daca237fd084b4345
Preparing worktree (detached HEAD 2f7285a72969)
Updating files: 100% (44080/44080), done.
HEAD is now at 2f7285a72969 Merge staging-next into staging
$ nix-env --extra-experimental-features no-url-literals --option system x86_64-darwin -f <nixpkgs> --nix-path nixpkgs=/Users/sirio/.cache/nixpkgs-review/pr-352129/nixpkgs nixpkgs-overlays=/var/folders/wf/hvl_z7t95xq489pb07g9sxzc0000gn/T/tmpetxhova4 -qaP --xml --out-path --show-trace --no-allow-import-from-derivation
error:
       … while evaluating the attribute 'blucontrol'
         at /Users/sirio/.cache/nixpkgs-review/pr-352129/nixpkgs/pkgs/top-level/all-packages.nix:28080:3:
        28079|
        28080|   blucontrol = callPackage ../applications/misc/blucontrol/wrapper.nix {
             |   ^
        28081|     inherit (haskellPackages) ghcWithPackages;

       … while evaluating the 'name' attribute of a derivation

       … from call site
         at /Users/sirio/.cache/nixpkgs-review/pr-352129/nixpkgs/pkgs/stdenv/generic/make-derivation.nix:354:9:
          353|         in
          354|         lib.strings.sanitizeDerivationName (
             |         ^
          355|           if attrs ? name

       … while calling anonymous lambda
         at /Users/sirio/.cache/nixpkgs-review/pr-352129/nixpkgs/lib/strings.nix:2625:3:
         2624|   in
         2625|   string:
             |   ^
         2626|   # First detect the common case of already valid strings, to speed those up

       … while evaluating a branch condition
         at /Users/sirio/.cache/nixpkgs-review/pr-352129/nixpkgs/lib/strings.nix:2627:3:
         2626|   # First detect the common case of already valid strings, to speed those up
         2627|   if stringLength string <= 207 && okRegex string != null
             |   ^
         2628|   then unsafeDiscardStringContext string

       … in the left operand of the AND (&&) operator
         at /Users/sirio/.cache/nixpkgs-review/pr-352129/nixpkgs/lib/strings.nix:2627:33:
         2626|   # First detect the common case of already valid strings, to speed those up
         2627|   if stringLength string <= 207 && okRegex string != null
             |                                 ^
         2628|   then unsafeDiscardStringContext string

       … in the argument of the not operator
         at /Users/sirio/.cache/nixpkgs-review/pr-352129/nixpkgs/lib/strings.nix:2627:26:
         2626|   # First detect the common case of already valid strings, to speed those up
         2627|   if stringLength string <= 207 && okRegex string != null
             |                          ^
         2628|   then unsafeDiscardStringContext string

       … while calling the 'lessThan' builtin
         at /Users/sirio/.cache/nixpkgs-review/pr-352129/nixpkgs/lib/strings.nix:2627:26:
         2626|   # First detect the common case of already valid strings, to speed those up
         2627|   if stringLength string <= 207 && okRegex string != null
             |                          ^
         2628|   then unsafeDiscardStringContext string

       … while calling the 'stringLength' builtin
         at /Users/sirio/.cache/nixpkgs-review/pr-352129/nixpkgs/lib/strings.nix:2627:6:
         2626|   # First detect the common case of already valid strings, to speed those up
         2627|   if stringLength string <= 207 && okRegex string != null
             |      ^
         2628|   then unsafeDiscardStringContext string

       … in the condition of the assert statement
         at /Users/sirio/.cache/nixpkgs-review/pr-352129/nixpkgs/pkgs/stdenv/generic/make-derivation.nix:359:13:
          358|             # we cannot coerce null to a string below
          359|             assert assertMsg (attrs ? version && attrs.version != null) "The `version` attribute cannot be null.";
             |             ^
          360|             "${attrs.pname}${staticMarker}${hostSuffix}-${attrs.version}"

       … from call site
         at /Users/sirio/.cache/nixpkgs-review/pr-352129/nixpkgs/pkgs/stdenv/generic/make-derivation.nix:359:20:
          358|             # we cannot coerce null to a string below
          359|             assert assertMsg (attrs ? version && attrs.version != null) "The `version` attribute cannot be null.";
             |                    ^
          360|             "${attrs.pname}${staticMarker}${hostSuffix}-${attrs.version}"

       … while calling 'assertMsg'
         at /Users/sirio/.cache/nixpkgs-review/pr-352129/nixpkgs/lib/asserts.nix:41:5:
           40|     pred:
           41|     msg:
             |     ^
           42|     pred || builtins.throw msg;

       … in the left operand of the OR (||) operator
         at /Users/sirio/.cache/nixpkgs-review/pr-352129/nixpkgs/lib/asserts.nix:42:10:
           41|     msg:
           42|     pred || builtins.throw msg;
             |          ^
           43|

       … in the right operand of the AND (&&) operator
         at /Users/sirio/.cache/nixpkgs-review/pr-352129/nixpkgs/pkgs/stdenv/generic/make-derivation.nix:359:47:
          358|             # we cannot coerce null to a string below
          359|             assert assertMsg (attrs ? version && attrs.version != null) "The `version` attribute cannot be null.";
             |                                               ^
          360|             "${attrs.pname}${staticMarker}${hostSuffix}-${attrs.version}"

       … while evaluating the attribute 'version'
         at /Users/sirio/.cache/nixpkgs-review/pr-352129/nixpkgs/pkgs/applications/misc/blucontrol/wrapper.nix:7:5:
            6|     pname = "blucontrol-with-packages";
            7|     version = blucontrolEnv.version;
             |     ^
            8|

       … while calling a functor (an attribute set with a '__functor' attribute)
         at /Users/sirio/.cache/nixpkgs-review/pr-352129/nixpkgs/pkgs/applications/misc/blucontrol/wrapper.nix:3:19:
            2| let
            3|   blucontrolEnv = ghcWithPackages (self: [ self.blucontrol ] ++ packages self);
             |                   ^
            4| in

       … from call site
         at /Users/sirio/.cache/nixpkgs-review/pr-352129/nixpkgs/pkgs/development/haskell-modules/make-package-set.nix:94:56:
           93|       # Converts a returned function to a functor attribute set if necessary
           94|       ensureAttrs = v: if builtins.isFunction v then { __functor = _: v; } else v;
             |                                                        ^
           95|

       … while calling anonymous lambda
         at /Users/sirio/.cache/nixpkgs-review/pr-352129/nixpkgs/pkgs/development/haskell-modules/with-packages-wrapper.nix:19:1:
           18| #   (hpkgs: [ hpkgs.mtl hpkgs.lens ])
           19| selectPackages:
             | ^
           20|

       … while evaluating a branch condition
         at /Users/sirio/.cache/nixpkgs-review/pr-352129/nixpkgs/pkgs/development/haskell-modules/with-packages-wrapper.nix:79:1:
           78|
           79| if paths == [] && !useLLVM then ghc else
             | ^
           80| symlinkJoin {

       … in the left operand of the AND (&&) operator
         at /Users/sirio/.cache/nixpkgs-review/pr-352129/nixpkgs/pkgs/development/haskell-modules/with-packages-wrapper.nix:79:16:
           78|
           79| if paths == [] && !useLLVM then ghc else
             |                ^
           80| symlinkJoin {

       … while calling the 'concatLists' builtin
         at /Users/sirio/.cache/nixpkgs-review/pr-352129/nixpkgs/pkgs/development/haskell-modules/with-packages-wrapper.nix:64:19:
           63|   packageCfgDir = "${libDir}/package.conf.d";
           64|   paths         = lib.concatLists (
             |                   ^
           65|                     builtins.map

       … while calling the 'map' builtin
         at /Users/sirio/.cache/nixpkgs-review/pr-352129/nixpkgs/pkgs/development/haskell-modules/with-packages-wrapper.nix:65:21:
           64|   paths         = lib.concatLists (
           65|                     builtins.map
             |                     ^
           66|                       (pkg: [ pkg ] ++ lib.optionals installDocumentation [ (lib.getOutput "doc" pkg) ])

       … while calling the 'filter' builtin
         at /Users/sirio/.cache/nixpkgs-review/pr-352129/nixpkgs/pkgs/development/haskell-modules/with-packages-wrapper.nix:67:24:
           66|                       (pkg: [ pkg ] ++ lib.optionals installDocumentation [ (lib.getOutput "doc" pkg) ])
           67|                       (lib.filter (x: x ? isHaskellLibrary) (lib.closePropagation packages))
             |                        ^
           68|                   );

       … from call site
         at /Users/sirio/.cache/nixpkgs-review/pr-352129/nixpkgs/pkgs/development/haskell-modules/with-packages-wrapper.nix:67:62:
           66|                       (pkg: [ pkg ] ++ lib.optionals installDocumentation [ (lib.getOutput "doc" pkg) ])
           67|                       (lib.filter (x: x ? isHaskellLibrary) (lib.closePropagation packages))
             |                                                              ^
           68|                   );

       … while calling 'closePropagationFast'
         at /Users/sirio/.cache/nixpkgs-review/pr-352129/nixpkgs/lib/deprecated/misc.nix:192:26:
          191|   # See https://github.com/NixOS/nixpkgs/pull/194391 for details.
          192|   closePropagationFast = list:
             |                          ^
          193|     builtins.map (x: x.val) (builtins.genericClosure {

       … while calling the 'map' builtin
         at /Users/sirio/.cache/nixpkgs-review/pr-352129/nixpkgs/lib/deprecated/misc.nix:193:5:
          192|   closePropagationFast = list:
          193|     builtins.map (x: x.val) (builtins.genericClosure {
             |     ^
          194|       startSet = builtins.map (x: {

       … while calling the 'genericClosure' builtin
         at /Users/sirio/.cache/nixpkgs-review/pr-352129/nixpkgs/lib/deprecated/misc.nix:193:30:
          192|   closePropagationFast = list:
          193|     builtins.map (x: x.val) (builtins.genericClosure {
             |                              ^
          194|       startSet = builtins.map (x: {

       … while evaluating the attribute 'outPath'
         at /Users/sirio/.cache/nixpkgs-review/pr-352129/nixpkgs/lib/customisation.nix:366:7:
          365|       drvPath = assert condition; drv.drvPath;
          366|       outPath = assert condition; drv.outPath;
             |       ^
          367|     };

       … while calling the 'getAttr' builtin
         at <nix/derivation-internal.nix>:44:19:
           43|       value = commonAttrs // {
           44|         outPath = builtins.getAttr outputName strict;
             |                   ^
           45|         drvPath = strict.drvPath;

       … while calling the 'derivationStrict' builtin
         at <nix/derivation-internal.nix>:34:12:
           33|
           34|   strict = derivationStrict drvAttrs;
             |            ^
           35|

       … while evaluating derivation 'blucontrol-0.7.1.0'
         whose name attribute is located at /Users/sirio/.cache/nixpkgs-review/pr-352129/nixpkgs/pkgs/stdenv/generic/make-derivation.nix:336:7

       … while evaluating attribute 'buildInputs' of derivation 'blucontrol-0.7.1.0'
         at /Users/sirio/.cache/nixpkgs-review/pr-352129/nixpkgs/pkgs/stdenv/generic/make-derivation.nix:383:7:
          382|       depsHostHost                = elemAt (elemAt dependencies 1) 0;
          383|       buildInputs                 = elemAt (elemAt dependencies 1) 1;
             |       ^
          384|       depsTargetTarget            = elemAt (elemAt dependencies 2) 0;

       … while calling the 'getAttr' builtin
         at <nix/derivation-internal.nix>:44:19:
           43|       value = commonAttrs // {
           44|         outPath = builtins.getAttr outputName strict;
             |                   ^
           45|         drvPath = strict.drvPath;

       … while calling the 'derivationStrict' builtin
         at <nix/derivation-internal.nix>:34:12:
           33|
           34|   strict = derivationStrict drvAttrs;
             |            ^
           35|

       … while evaluating derivation 'libX11-1.8.10'
         whose name attribute is located at /Users/sirio/.cache/nixpkgs-review/pr-352129/nixpkgs/pkgs/stdenv/generic/make-derivation.nix:336:7

       … while evaluating attribute 'buildInputs' of derivation 'libX11-1.8.10'
         at /Users/sirio/.cache/nixpkgs-review/pr-352129/nixpkgs/pkgs/stdenv/generic/make-derivation.nix:383:7:
          382|       depsHostHost                = elemAt (elemAt dependencies 1) 0;
          383|       buildInputs                 = elemAt (elemAt dependencies 1) 1;
             |       ^
          384|       depsTargetTarget            = elemAt (elemAt dependencies 2) 0;

       … while calling the 'getAttr' builtin
         at <nix/derivation-internal.nix>:44:19:
           43|       value = commonAttrs // {
           44|         outPath = builtins.getAttr outputName strict;
             |                   ^
           45|         drvPath = strict.drvPath;

       … while calling the 'derivationStrict' builtin
         at <nix/derivation-internal.nix>:34:12:
           33|
           34|   strict = derivationStrict drvAttrs;
             |            ^
           35|

       … while evaluating derivation 'xorgproto-2024.1'
         whose name attribute is located at /Users/sirio/.cache/nixpkgs-review/pr-352129/nixpkgs/pkgs/stdenv/generic/make-derivation.nix:336:7

       … while evaluating attribute '__impureHostDeps' of derivation 'xorgproto-2024.1'
         at /Users/sirio/.cache/nixpkgs-review/pr-352129/nixpkgs/pkgs/stdenv/generic/make-derivation.nix:451:7:
          450|       __propagatedSandboxProfile = unique (computedPropagatedSandboxProfile ++ [ propagatedSandboxProfile ]);
          451|       __impureHostDeps = computedImpureHostDeps ++ computedPropagatedImpureHostDeps ++ __propagatedImpureHostDeps ++ __impureHostDeps ++ stdenv.__extraImpureHostDeps ++ [
             |       ^
          452|         "/dev/zero"

       … while calling the 'foldl'' builtin
         at /Users/sirio/.cache/nixpkgs-review/pr-352129/nixpkgs/pkgs/stdenv/generic/make-derivation.nix:435:11:
          434|         computedImpureHostDeps =
          435|           unique (concatMap (input: input.__propagatedImpureHostDeps or [])
             |           ^
          436|             (stdenv.extraNativeBuildInputs

       … while calling the 'concatMap' builtin
         at /Users/sirio/.cache/nixpkgs-review/pr-352129/nixpkgs/pkgs/stdenv/generic/make-derivation.nix:435:19:
          434|         computedImpureHostDeps =
          435|           unique (concatMap (input: input.__propagatedImpureHostDeps or [])
             |                   ^
          436|             (stdenv.extraNativeBuildInputs

       … while calling anonymous lambda
         at /Users/sirio/.cache/nixpkgs-review/pr-352129/nixpkgs/pkgs/stdenv/generic/make-derivation.nix:435:30:
          434|         computedImpureHostDeps =
          435|           unique (concatMap (input: input.__propagatedImpureHostDeps or [])
             |                              ^
          436|             (stdenv.extraNativeBuildInputs

       … while evaluating the attribute '__propagatedImpureHostDeps'
         at /Users/sirio/.cache/nixpkgs-review/pr-352129/nixpkgs/pkgs/stdenv/generic/make-derivation.nix:457:7:
          456|       ];
          457|       __propagatedImpureHostDeps = computedPropagatedImpureHostDeps ++ __propagatedImpureHostDeps;
             |       ^
          458|     }) //

       … while calling the 'foldl'' builtin
         at /Users/sirio/.cache/nixpkgs-review/pr-352129/nixpkgs/pkgs/stdenv/generic/make-derivation.nix:441:11:
          440|         computedPropagatedImpureHostDeps =
          441|           unique (concatMap (input: input.__propagatedImpureHostDeps or [])
             |           ^
          442|             allPropagatedDependencies);

       … while calling the 'concatMap' builtin
         at /Users/sirio/.cache/nixpkgs-review/pr-352129/nixpkgs/pkgs/stdenv/generic/make-derivation.nix:441:19:
          440|         computedPropagatedImpureHostDeps =
          441|           unique (concatMap (input: input.__propagatedImpureHostDeps or [])
             |                   ^
          442|             allPropagatedDependencies);

       … while calling anonymous lambda
         at /Users/sirio/.cache/nixpkgs-review/pr-352129/nixpkgs/pkgs/stdenv/generic/make-derivation.nix:441:30:
          440|         computedPropagatedImpureHostDeps =
          441|           unique (concatMap (input: input.__propagatedImpureHostDeps or [])
             |                              ^
          442|             allPropagatedDependencies);

       … from call site
         at /Users/sirio/.cache/nixpkgs-review/pr-352129/nixpkgs/pkgs/stdenv/generic/make-derivation.nix:441:37:
          440|         computedPropagatedImpureHostDeps =
          441|           unique (concatMap (input: input.__propagatedImpureHostDeps or [])
             |                                     ^
          442|             allPropagatedDependencies);

       … while calling anonymous lambda
         at /Users/sirio/.cache/nixpkgs-review/pr-352129/nixpkgs/pkgs/stdenv/generic/make-derivation.nix:326:13:
          325|       (map (drv: getDev drv.__spliced.hostHost or drv) (checkDependencyList "depsHostHostPropagated" depsHostHostPropagated))
          326|       (map (drv: getDev drv.__spliced.hostTarget or drv) (checkDependencyList "propagatedBuildInputs" propagatedBuildInputs))
             |             ^
          327|     ]

       … from call site
         at /Users/sirio/.cache/nixpkgs-review/pr-352129/nixpkgs/pkgs/stdenv/generic/make-derivation.nix:326:18:
          325|       (map (drv: getDev drv.__spliced.hostHost or drv) (checkDependencyList "depsHostHostPropagated" depsHostHostPropagated))
          326|       (map (drv: getDev drv.__spliced.hostTarget or drv) (checkDependencyList "propagatedBuildInputs" propagatedBuildInputs))
             |                  ^
          327|     ]

       … while calling 'getOutput'
         at /Users/sirio/.cache/nixpkgs-review/pr-352129/nixpkgs/lib/attrsets.nix:1796:23:
         1795|   */
         1796|   getOutput = output: pkg:
             |                       ^
         1797|     if ! pkg ? outputSpecified || ! pkg.outputSpecified

       … while evaluating a branch condition
         at /Users/sirio/.cache/nixpkgs-review/pr-352129/nixpkgs/lib/attrsets.nix:1797:5:
         1796|   getOutput = output: pkg:
         1797|     if ! pkg ? outputSpecified || ! pkg.outputSpecified
             |     ^
         1798|       then pkg.${output} or pkg.out or pkg

       … in the left operand of the OR (||) operator
         at /Users/sirio/.cache/nixpkgs-review/pr-352129/nixpkgs/lib/attrsets.nix:1797:32:
         1796|   getOutput = output: pkg:
         1797|     if ! pkg ? outputSpecified || ! pkg.outputSpecified
             |                                ^
         1798|       then pkg.${output} or pkg.out or pkg

       … in the argument of the not operator
         at /Users/sirio/.cache/nixpkgs-review/pr-352129/nixpkgs/lib/attrsets.nix:1797:10:
         1796|   getOutput = output: pkg:
         1797|     if ! pkg ? outputSpecified || ! pkg.outputSpecified
             |          ^
         1798|       then pkg.${output} or pkg.out or pkg

       … from call site
         at /Users/sirio/.cache/nixpkgs-review/pr-352129/nixpkgs/pkgs/stdenv/generic/make-derivation.nix:326:25:
          325|       (map (drv: getDev drv.__spliced.hostHost or drv) (checkDependencyList "depsHostHostPropagated" depsHostHostPropagated))
          326|       (map (drv: getDev drv.__spliced.hostTarget or drv) (checkDependencyList "propagatedBuildInputs" propagatedBuildInputs))
             |                         ^
          327|     ]

       … while calling anonymous lambda
         at /Users/sirio/.cache/nixpkgs-review/pr-352129/nixpkgs/lib/lists.nix:334:29:
          333|   */
          334|   imap1 = f: list: genList (n: f (n + 1) (elemAt list n)) (length list);
             |                             ^
          335|

       … from call site
         at /Users/sirio/.cache/nixpkgs-review/pr-352129/nixpkgs/lib/lists.nix:334:32:
          333|   */
          334|   imap1 = f: list: genList (n: f (n + 1) (elemAt list n)) (length list);
             |                                ^
          335|

       … while calling anonymous lambda
         at /Users/sirio/.cache/nixpkgs-review/pr-352129/nixpkgs/pkgs/stdenv/generic/make-derivation.nix:281:15:
          280|     imap1
          281|       (index: dep:
             |               ^
          282|         if isDerivation dep || dep == null || builtins.isString dep || builtins.isPath dep then dep

       … while evaluating a branch condition
         at /Users/sirio/.cache/nixpkgs-review/pr-352129/nixpkgs/pkgs/stdenv/generic/make-derivation.nix:282:9:
          281|       (index: dep:
          282|         if isDerivation dep || dep == null || builtins.isString dep || builtins.isPath dep then dep
             |         ^
          283|         else if isList dep then checkDependencyList' ([index] ++ positions) name dep

       … in the left operand of the OR (||) operator
         at /Users/sirio/.cache/nixpkgs-review/pr-352129/nixpkgs/pkgs/stdenv/generic/make-derivation.nix:282:69:
          281|       (index: dep:
          282|         if isDerivation dep || dep == null || builtins.isString dep || builtins.isPath dep then dep
             |                                                                     ^
          283|         else if isList dep then checkDependencyList' ([index] ++ positions) name dep

       … in the left operand of the OR (||) operator
         at /Users/sirio/.cache/nixpkgs-review/pr-352129/nixpkgs/pkgs/stdenv/generic/make-derivation.nix:282:44:
          281|       (index: dep:
          282|         if isDerivation dep || dep == null || builtins.isString dep || builtins.isPath dep then dep
             |                                            ^
          283|         else if isList dep then checkDependencyList' ([index] ++ positions) name dep

       … in the left operand of the OR (||) operator
         at /Users/sirio/.cache/nixpkgs-review/pr-352129/nixpkgs/pkgs/stdenv/generic/make-derivation.nix:282:29:
          281|       (index: dep:
          282|         if isDerivation dep || dep == null || builtins.isString dep || builtins.isPath dep then dep
             |                             ^
          283|         else if isList dep then checkDependencyList' ([index] ++ positions) name dep

       … from call site
         at /Users/sirio/.cache/nixpkgs-review/pr-352129/nixpkgs/pkgs/stdenv/generic/make-derivation.nix:282:12:
          281|       (index: dep:
          282|         if isDerivation dep || dep == null || builtins.isString dep || builtins.isPath dep then dep
             |            ^
          283|         else if isList dep then checkDependencyList' ([index] ++ positions) name dep

       … while calling 'isDerivation'
         at /Users/sirio/.cache/nixpkgs-review/pr-352129/nixpkgs/lib/attrsets.nix:1282:5:
         1281|   isDerivation =
         1282|     value: value.type or null == "derivation";
             |     ^
         1283|

       … while calling the 'elemAt' builtin
         at /Users/sirio/.cache/nixpkgs-review/pr-352129/nixpkgs/lib/lists.nix:334:43:
          333|   */
          334|   imap1 = f: list: genList (n: f (n + 1) (elemAt list n)) (length list);
             |                                           ^
          335|

       … while calling 'checkDrv'
         at /Users/sirio/.cache/nixpkgs-review/pr-352129/nixpkgs/pkgs/development/interpreters/python/mk-python-derivation.nix:206:26:
          205|
          206|     checkDrv = attrName: drv:
             |                          ^
          207|       if (isPythonModule drv) && (isMismatchedPython drv) then throwMismatch attrName drv

       … while evaluating a branch condition
         at /Users/sirio/.cache/nixpkgs-review/pr-352129/nixpkgs/pkgs/development/interpreters/python/mk-python-derivation.nix:207:7:
          206|     checkDrv = attrName: drv:
          207|       if (isPythonModule drv) && (isMismatchedPython drv) then throwMismatch attrName drv
             |       ^
          208|       else drv;

       … in the right operand of the AND (&&) operator
         at /Users/sirio/.cache/nixpkgs-review/pr-352129/nixpkgs/pkgs/development/interpreters/python/mk-python-derivation.nix:207:31:
          206|     checkDrv = attrName: drv:
          207|       if (isPythonModule drv) && (isMismatchedPython drv) then throwMismatch attrName drv
             |                               ^
          208|       else drv;

       … from call site
         at /Users/sirio/.cache/nixpkgs-review/pr-352129/nixpkgs/pkgs/development/interpreters/python/mk-python-derivation.nix:207:35:
          206|     checkDrv = attrName: drv:
          207|       if (isPythonModule drv) && (isMismatchedPython drv) then throwMismatch attrName drv
             |                                   ^
          208|       else drv;

       … while calling 'isMismatchedPython'
         at /Users/sirio/.cache/nixpkgs-review/pr-352129/nixpkgs/pkgs/development/interpreters/python/mk-python-derivation.nix:48:24:
           47|
           48|   isMismatchedPython = drv: drv.pythonModule != python;
             |                        ^
           49|

       … while calling the 'getAttr' builtin
         at <nix/derivation-internal.nix>:44:19:
           43|       value = commonAttrs // {
           44|         outPath = builtins.getAttr outputName strict;
             |                   ^
           45|         drvPath = strict.drvPath;

       … while calling the 'derivationStrict' builtin
         at <nix/derivation-internal.nix>:34:12:
           33|
           34|   strict = derivationStrict drvAttrs;
             |            ^
           35|

       … while evaluating derivation 'python3-3.12.7'
         whose name attribute is located at /Users/sirio/.cache/nixpkgs-review/pr-352129/nixpkgs/pkgs/stdenv/generic/make-derivation.nix:336:7

       … while evaluating attribute 'CPPFLAGS' of derivation 'python3-3.12.7'

       … while calling anonymous lambda
         at /Users/sirio/.cache/nixpkgs-review/pr-352129/nixpkgs/pkgs/stdenv/generic/make-derivation.nix:584:11:
          583|     mapAttrs
          584|       (n: v: assert assertMsg (isString v || isBool v || isInt v || isDerivation v)
             |           ^
          585|         "The `env` attribute set can only contain derivation, string, boolean or integer attributes. The `${n}` attribute is of type ${builtins.typeOf v}."; v)

       … in the condition of the assert statement
         at /Users/sirio/.cache/nixpkgs-review/pr-352129/nixpkgs/pkgs/stdenv/generic/make-derivation.nix:584:14:
          583|     mapAttrs
          584|       (n: v: assert assertMsg (isString v || isBool v || isInt v || isDerivation v)
             |              ^
          585|         "The `env` attribute set can only contain derivation, string, boolean or integer attributes. The `${n}` attribute is of type ${builtins.typeOf v}."; v)

       … from call site
         at /Users/sirio/.cache/nixpkgs-review/pr-352129/nixpkgs/pkgs/stdenv/generic/make-derivation.nix:584:21:
          583|     mapAttrs
          584|       (n: v: assert assertMsg (isString v || isBool v || isInt v || isDerivation v)
             |                     ^
          585|         "The `env` attribute set can only contain derivation, string, boolean or integer attributes. The `${n}` attribute is of type ${builtins.typeOf v}."; v)

       … while calling 'assertMsg'
         at /Users/sirio/.cache/nixpkgs-review/pr-352129/nixpkgs/lib/asserts.nix:41:5:
           40|     pred:
           41|     msg:
             |     ^
           42|     pred || builtins.throw msg;

       … in the left operand of the OR (||) operator
         at /Users/sirio/.cache/nixpkgs-review/pr-352129/nixpkgs/lib/asserts.nix:42:10:
           41|     msg:
           42|     pred || builtins.throw msg;
             |          ^
           43|

       … in the left operand of the OR (||) operator
         at /Users/sirio/.cache/nixpkgs-review/pr-352129/nixpkgs/pkgs/stdenv/generic/make-derivation.nix:584:66:
          583|     mapAttrs
          584|       (n: v: assert assertMsg (isString v || isBool v || isInt v || isDerivation v)
             |                                                                  ^
          585|         "The `env` attribute set can only contain derivation, string, boolean or integer attributes. The `${n}` attribute is of type ${builtins.typeOf v}."; v)

       … in the left operand of the OR (||) operator
         at /Users/sirio/.cache/nixpkgs-review/pr-352129/nixpkgs/pkgs/stdenv/generic/make-derivation.nix:584:55:
          583|     mapAttrs
          584|       (n: v: assert assertMsg (isString v || isBool v || isInt v || isDerivation v)
             |                                                       ^
          585|         "The `env` attribute set can only contain derivation, string, boolean or integer attributes. The `${n}` attribute is of type ${builtins.typeOf v}."; v)

       … in the left operand of the OR (||) operator
         at /Users/sirio/.cache/nixpkgs-review/pr-352129/nixpkgs/pkgs/stdenv/generic/make-derivation.nix:584:43:
          583|     mapAttrs
          584|       (n: v: assert assertMsg (isString v || isBool v || isInt v || isDerivation v)
             |                                           ^
          585|         "The `env` attribute set can only contain derivation, string, boolean or integer attributes. The `${n}` attribute is of type ${builtins.typeOf v}."; v)

       … while calling the 'isString' builtin
         at /Users/sirio/.cache/nixpkgs-review/pr-352129/nixpkgs/pkgs/stdenv/generic/make-derivation.nix:584:32:
          583|     mapAttrs
          584|       (n: v: assert assertMsg (isString v || isBool v || isInt v || isDerivation v)
             |                                ^
          585|         "The `env` attribute set can only contain derivation, string, boolean or integer attributes. The `${n}` attribute is of type ${builtins.typeOf v}."; v)

       … while calling the 'concatStringsSep' builtin
         at /Users/sirio/.cache/nixpkgs-review/pr-352129/nixpkgs/pkgs/development/interpreters/python/cpython/default.nix:329:16:
          328|   env = {
          329|     CPPFLAGS = concatStringsSep " " (map (p: "-I${getDev p}/include") buildInputs);
             |                ^
          330|     LDFLAGS = concatStringsSep " " (map (p: "-L${getLib p}/lib") buildInputs);

       … while calling anonymous lambda
         at /Users/sirio/.cache/nixpkgs-review/pr-352129/nixpkgs/pkgs/development/interpreters/python/cpython/default.nix:329:43:
          328|   env = {
          329|     CPPFLAGS = concatStringsSep " " (map (p: "-I${getDev p}/include") buildInputs);
             |                                           ^
          330|     LDFLAGS = concatStringsSep " " (map (p: "-L${getLib p}/lib") buildInputs);

       … while calling the 'getAttr' builtin
         at <nix/derivation-internal.nix>:44:19:
           43|       value = commonAttrs // {
           44|         outPath = builtins.getAttr outputName strict;
             |                   ^
           45|         drvPath = strict.drvPath;

       … while calling the 'derivationStrict' builtin
         at <nix/derivation-internal.nix>:34:12:
           33|
           34|   strict = derivationStrict drvAttrs;
             |            ^
           35|

       … while evaluating derivation 'apple-sdk-11.3'
         whose name attribute is located at /Users/sirio/.cache/nixpkgs-review/pr-352129/nixpkgs/pkgs/stdenv/generic/make-derivation.nix:336:7

       … while evaluating attribute 'propagatedBuildInputs' of derivation 'apple-sdk-11.3'
         at /Users/sirio/.cache/nixpkgs-review/pr-352129/nixpkgs/pkgs/stdenv/generic/make-derivation.nix:390:7:
          389|       depsHostHostPropagated      = elemAt (elemAt propagatedDependencies 1) 0;
          390|       propagatedBuildInputs       = elemAt (elemAt propagatedDependencies 1) 1;
             |       ^
          391|       depsTargetTargetPropagated  = elemAt (elemAt propagatedDependencies 2) 0;

       … while evaluating list element at index 3

       … while calling the 'getAttr' builtin
         at <nix/derivation-internal.nix>:44:19:
           43|       value = commonAttrs // {
           44|         outPath = builtins.getAttr outputName strict;
             |                   ^
           45|         drvPath = strict.drvPath;

       … while calling the 'derivationStrict' builtin
         at <nix/derivation-internal.nix>:34:12:
           33|
           34|   strict = derivationStrict drvAttrs;
             |            ^
           35|

       … while evaluating derivation 'libutil-72'
         whose name attribute is located at /Users/sirio/.cache/nixpkgs-review/pr-352129/nixpkgs/pkgs/stdenv/generic/make-derivation.nix:336:7

       … while evaluating attribute '__impureHostDeps' of derivation 'libutil-72'
         at /Users/sirio/.cache/nixpkgs-review/pr-352129/nixpkgs/pkgs/stdenv/generic/make-derivation.nix:451:7:
          450|       __propagatedSandboxProfile = unique (computedPropagatedSandboxProfile ++ [ propagatedSandboxProfile ]);
          451|       __impureHostDeps = computedImpureHostDeps ++ computedPropagatedImpureHostDeps ++ __propagatedImpureHostDeps ++ __impureHostDeps ++ stdenv.__extraImpureHostDeps ++ [
             |       ^
          452|         "/dev/zero"

       … while calling the 'foldl'' builtin
         at /Users/sirio/.cache/nixpkgs-review/pr-352129/nixpkgs/pkgs/stdenv/generic/make-derivation.nix:435:11:
          434|         computedImpureHostDeps =
          435|           unique (concatMap (input: input.__propagatedImpureHostDeps or [])
             |           ^
          436|             (stdenv.extraNativeBuildInputs

       … while calling the 'concatMap' builtin
         at /Users/sirio/.cache/nixpkgs-review/pr-352129/nixpkgs/pkgs/stdenv/generic/make-derivation.nix:435:19:
          434|         computedImpureHostDeps =
          435|           unique (concatMap (input: input.__propagatedImpureHostDeps or [])
             |                   ^
          436|             (stdenv.extraNativeBuildInputs

       … while calling anonymous lambda
         at /Users/sirio/.cache/nixpkgs-review/pr-352129/nixpkgs/pkgs/stdenv/generic/make-derivation.nix:435:30:
          434|         computedImpureHostDeps =
          435|           unique (concatMap (input: input.__propagatedImpureHostDeps or [])
             |                              ^
          436|             (stdenv.extraNativeBuildInputs

       … while evaluating the attribute '__propagatedImpureHostDeps'
         at /Users/sirio/.cache/nixpkgs-review/pr-352129/nixpkgs/pkgs/stdenv/generic/make-derivation.nix:457:7:
          456|       ];
          457|       __propagatedImpureHostDeps = computedPropagatedImpureHostDeps ++ __propagatedImpureHostDeps;
             |       ^
          458|     }) //

       error: infinite recursion encountered
       at /Users/sirio/.cache/nixpkgs-review/pr-352129/nixpkgs/pkgs/stdenv/generic/make-derivation.nix:457:7:
          456|       ];
          457|       __propagatedImpureHostDeps = computedPropagatedImpureHostDeps ++ __propagatedImpureHostDeps;
             |       ^
          458|     }) //
https://github.com/NixOS/nixpkgs/pull/352129 failed to build: Failed to list packages: nix-env failed with exit code 1
$ git worktree remove -f /Users/sirio/.cache/nixpkgs-review/pr-352129/nixpkgs

Error: exit status 1
```

emilazy commented:
That’s an unrelated eval issue that I’ve just fixed, sorry! Rebased accordingly.

That said, this will rebuild literally every package, so running a full `nixpkgs-review` cycle is somewhat hopeless :) (But it can still help check for eval problems!)

paparodeo commented on pkgs/os-specific/darwin/apple-source-releases/file_cmds/package.nix:
```diff
@@ -1,11 +1,9 @@
 {
   lib,
   apple-sdk,
-  apple-sdk_11,
```
file_cmds is failing to build for me due to being unable to find the `sha.h` header.

reckenrode commented on pkgs/development/compilers/llvm/common/compiler-rt/default.nix:
```diff
@@ -47,13 +46,7 @@ let
-      apple-sdk_10_13.override { enableBootstrap = true; }
-    else
-      apple-sdk.override { enableBootstrap = true; };
+  apple-sdk' = apple-sdk.override { enableBootstrap = true; };
```
This should be unnecessary because compiler-rt is built with the bootstrap stdenv on Darwin. This stuff was a workaround for the fact the 10.12 SDK is just too old now.

emilazy replied:
Fixed, and confirmed that it still bootstraps.

emilazy commented:
Okay, I’ve confirmed that bootstrap continues to work after actually pushing the `compiler-rt` change that I had previously neglected to. I’ve tested builds pretty extensively on `x86_64-darwin` when I first put up the PR, and the LLVM 19 work that is based on this PR has resulted in a bunch more testing on `aarch64-darwin`. I’ve also tested the LLVM 19 branch on Linux, so this shouldn’t break eval or anything there. I haven’t heard any objections to aligning our support policy with Apple’s since putting this PR up, so I think it’s time to land this 🎉
